app/models/spree/gift_card.rb

```diff
@@ -63,6 +61,10 @@ def order_activatable?(order)
       !UNACTIVATABLE_ORDER_STATES.include?(order.state)
     end
 
+    def calculator
```

Comment:
@sawangupta92 - I guess this will create a new object of GiftCardCalculator every time calculator is called.